src/ln/channelmonitor.rs

```diff
@@ -1407,6 +1407,10 @@ impl ChannelMonitor {
 						output: spend_tx.output[0].clone(),
 					});
 					txn_to_broadcast.push(spend_tx);
+
+					// TODO: We need to fail back HTLCs that were't included in the broadcast
```

Comment:
If we create a transaction for the remote side, we *must* provide_latest_remote_tx (and update the monitor, and write it all to disk) before we send the signatures required for the remote to broadcast it. For local, we either have to succeed at committing new monitors or we have to fail the channel immediately using the current in-memory monitor (though I think we need to move to failing HTLCs based only on remotes, so this wont matter anyway).

Reply:
To clarify my question, what if remote peer broadcast its commitment tx while we're done with inbounds htlcs (so we have them in HTLCForwardInfo after Channel::forward_htlcs) but before we call process_pending_htlc_forwards ? I didn't test it, but seems we're unable to fail backward inbound HTLCs...

Reply:
Then the counterparty would be failing a transaction with inbound HTLCs, not outbound ones, which we aren't really handling in this patch. If they dutifully committed to an HTLC (implying the corresponding ChannelMonitor was updated) and it ended up in our pending HTLC forwards there is nothing wrong with forwarding it, it just means when the upstream node either fails or forwards we need to get that information back to the original ChannelMonitor so it can do the same on chain. Not forwarding it may be nicer since we could avoid the on-chain fee and just let them claim the timeout, but that is a relatively small edge case and adding code to handle it probably isnt worth it.

Reply:
Okay good, well letting other pay the fee would be a nice extension of what a greedy algorithm means.